noxfile.py

```diff
@@ -0,0 +1,152 @@
+    "atmosphere/subgrid_scale_physics/microphysics",
+    "common",
+    "driver",
+    # "testing", #TODO: Add tests to testing subpackage
+]
+ModelTestsSubset: TypeAlias = Literal["datatest", "stencils", "basic"]
+
+MODEL_SUBPACKAGE_PATHS: Final[Sequence[str]] = ModelSubpackagePath.__args__
+MODEL_TESTS_SUBSETS: Final[Sequence[str]] = ModelTestsSubset.__args__
+
+# -- nox configuration --
+nox.options.default_venv_backend = "uv"
+nox.options.sessions = ["test_model", "test_tools"]
+
+
+# -- nox sessions --
+# Model benchmark sessions
+# TODO(egparedes): Add backend parameter
+# TODO(edopao,egparedes): Change 'extras' back to 'all' once mpi4py can be compiled with hpc_sdk
+@nox.session(python=["3.10", "3.11"])
+@nox.parametrize("subpackage", MODEL_SUBPACKAGE_PATHS)
+def benchmark_model(session: nox.Session, subpackage: ModelSubpackagePath) -> None:
+    """Run pytest benchmarks for selected icon4py model subpackages."""
+    _install_session_venv(session, extras=["dace", "io", "testing"], groups=["test"])
+
+    with session.chdir(f"model/{subpackage}"):
+        session.run(*"pytest -sv --benchmark-only".split(), *session.posargs)
+
+
+# Model test sessions
+# TODO(egparedes): Add backend parameter
+# TODO(edopao,egparedes): Change 'extras' back to 'all' once mpi4py can be compiled with hpc_sdk
+@nox.session(python=["3.10", "3.11"])
+@nox.parametrize("subpackage", MODEL_SUBPACKAGE_PATHS)
+@nox.parametrize("selection", MODEL_TESTS_SUBSETS)
+def test_model(session: nox.Session, selection: ModelTestsSubset, subpackage: ModelSubpackagePath) -> None:
+    """Run tests for selected icon4py model subpackages."""
+    _install_session_venv(session, extras=["dace", "fortran", "io", "testing"], groups=["test"])
+
+    pytest_args = _selection_to_pytest_args(selection)
+    with session.chdir(f"model/{subpackage}"):
+        session.run(
+            *f"pytest -sv --benchmark-skip -n {session.env.get('NUM_PROCESSES', 'auto')}".split(),
+            *pytest_args,
+            *session.posargs
+        )
+
+@nox.session(python=["3.10", "3.11"])
+@nox.parametrize("subpackage", MODEL_SUBPACKAGE_PATHS)
+def test_model_datatest(session: nox.Session, subpackage: ModelSubpackagePath) -> None:
+    session.notify(
+        f"test_model-{session.python}(selection='datatest', subpackage='{subpackage}')"
+    )
+
+
+@nox.session(python=["3.10", "3.11"])
+@nox.parametrize("subpackage", MODEL_SUBPACKAGE_PATHS)
+def test_model_stencils(session: nox.Session, subpackage: ModelSubpackagePath) -> None:
+    notest_subpackages = {  # test discovery fails because no stencil tests found
+        "atmosphere/subgrid_scale_physics/microphysics",
+        "driver",
+    }
+    if subpackage in notest_subpackages:
+        session.skip(f"no tests configured for {subpackage} subpackage")
+    elif subpackage != "common":  # TODO: Enable tests
+        session.notify(
+            f"test_model-{session.python}(selection='stencils', subpackage='{subpackage}')"
+        )
```

Comment:
```suggestion
    elif subpackage == "common":
        session.skip(f"tests broken")  # TODO: Enable tests
    session.notify(
            f"test_model-{session.python}(selection='stencils', subpackage='{subpackage}')"
        )
```
Probably this pattern?

Still not 100% sure what's the difference to main for this test set. I am not able to reproduce here https://github.com/C2SM/icon4py/pull/629 because the test set is empty.